labs/LanguageBasics/LanguageBasics.md

```diff
@@ -211,60 +625,137 @@ encrypt key plainText = cipherText
-* `>>$` is the signed right shift operator, a.k.a. arithmetic right shift
-* `/$` and `%$` are signed division and remainder operators
+Less Common Operators
+---------------------
 
-## Don't let the type system do your work
+Function equality: `===` and `!==`. These are mostly used to state
+properties about functions over a finite domain.
 
-Cryptol's type system tries to infer the types of functions lacking a type signature. Sometimes it comes up with a more general type than you were imagining. This causes problems:
-* Perhaps you only want your function to be applicable on a smaller set of types (usually minor, but occasionally major).
-* Error messages cab become even more incomprehensible! (Major)
+```
+add8 : [4] -> [4]
+add8 x = x + 8
+sub8 : [4] -> [4]
+sub8 x = x - 8
+```
 
-## Do let the type system work for you
+```shell
+labs::LanguageBasics::LanguageBasics> :prove add8 === sub8
+Q.E.D.
+(Total Elapsed Time: 0.014s, using Z3)
+```
+
+There is also an `:exhaust` command for finite domains. On occasion
+the machinery behind `:prove` gets overwhelmed and, on small enough
+domains, exhausting works in a reasonable time.
 
-Type signatures for functions are wonderful bits of documentation. It is much easier to see what's going on if you use type synonyms and signatures. 
+```shell
+labs::LanguageBasics::LanguageBasics> :exhaust add8 === sub8
+Using exhaustive testing.
+Passed 16 tests.
+Q.E.D.
+```
+
+The `:check` command is smart enough to notice small enough domains
+and switch to exhaustion automagically:
+
+```shell
+labs::LanguageBasics::LanguageBasics> :check add8 === sub8
+Using exhaustive testing.
+Passed 16 tests.
+Q.E.D.
+```
+
+
+Judicious Type System Usage
+---------------------------
+
+### Don't let the type system do your work
+
+Cryptol's type system tries to infer the types of functions lacking a
+type signature. Sometimes it comes up with a more general type than
+you were imagining. This causes problems:
+
+* Perhaps you only want your function to be applicable on a smaller
+  set of types (usually minor, but occasionally major).
+* Error messages can become even more incomprehensible! (Major)
+
+### Do let the type system work for you
+
+Type signatures for functions are wonderful bits of documentation. It
+is much easier to see what's going on if you use type synonyms and
+signatures.
 
 * makes cleaner code
 * easier for other tools to consume/reason about
 
-## Supporting code
+### Provide additional types to aid in debugging
+
+Many of the errors in coding Cryptol will be instances of type
+mismatching. If you can't see your problem based on the error message,
+try adding more type annotations. This
+* makes the interpreter do less work trying alternative possibilites
+and, consequently, can make error messages more comprehensible
+* reduces the body of code to examine for bugs (a sort of binary bug
+search)
+* can get you to notice where you blew it
```

Comment:
The "...annotations.  This" comes across as a typographical error.  Perhaps distribute "this"/"it"/"type annotations" to each bullet instead.

Reply:
How about just a colon after "This"?